### PR TITLE
Cli reorg

### PR DESCRIPTION
**Pull request by gila**
Rework some of the gRPC bits, and make a start, with adding non (directly) mayastor related methods that can help us inspect, test, setup etc during test.

### CODE REVIEW

File: mayastor/src/bin/cli/pool_cli.rs [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+use tonic::{Code, Status};
+
+pub fn subcommands<'a, 'b>() -> App<'a, 'b> {
+    let create = SubCommand::with_name("create")
+        .about("Create storage pool")
+        .arg(
+            Arg::with_name("block-size")
+                .short("b")
+                .long("block-size")
+                .value_name("NUMBER")
```
Comment (paulyoong) [inline mayastor/src/bin/cli/pool_cli.rs:14]: Should value name be size?

File: mayastor/src/bin/cli/cli.rs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                .short("a")
+                .long("address")
+                .default_value("127.0.0.1")
+                .value_name("HOST")
+                .help("IP address of mayastor instance"))
+        .arg(
+            Arg::with_name("port")
+                .short("p")
+                .long("port")
+                .default_value("10124").value_name("NUMBER")
```
[the remark this answers is not in the document]
Comment (gila) [reply mayastor/src/bin/cli/cli.rs:50]: the NUMBER indicates whatever portnumner you provide, this seems to be OK and common.

File: mayastor/src/bin/cli/pool_cli.rs (hunk 1 above)
Comment (gila) [reply mayastor/src/bin/cli/pool_cli.rs:14]: the size is a number...  I see your thinking, but this was as it was before and I've not touched the usage messages etc..

Comment (paulyoong) [reply mayastor/src/bin/cli/pool_cli.rs:14]: Ok :+1:

File: mayastor/src/bin/cli/cli.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (paulyoong) [reply mayastor/src/bin/cli/cli.rs:50]: :+1:

File: mayastor/src/bin/cli/context.rs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        } else {
+            matches.occurrences_of("verbose") + 1
+        };
+        let units = matches
+            .value_of("units")
+            .and_then(|u| u.chars().next())
+            .unwrap_or('b');
+        let endpoint = {
+            let addr = matches.value_of("address").unwrap_or("127.0.0.1");
+            let port = value_t!(matches.value_of("port"), u16).unwrap_or(10124);
```
[the remark this answers is not in the document]
Comment (gila) [reply mayastor/src/bin/cli/context.rs:26]: We do not I would agree, brushing up the argument handeling i would like to defer to later for now.

**Pull request merged**